mode/factor/factor.js

```diff
@@ -22,50 +22,62 @@
+      {regex: /\S*?"/, token: "string", next: "string"},
       // numbers: dec, hex, unicode, bin, fractional, complex
-      {regex: /(?:[+-]?)(?:0x[\d,a-f]+)|(?:0o[0-7]+)|(?:0b[0,1]+)|(?:\d+.?\d*)/, token: "number"},
+      {regex: /(?:0x[\d,a-f]+)|(?:0o[0-7]+)|(?:0b[0,1]+)|(?:\-?\d+.?\d*)(?=\s)/, token: "number"},
       //{regex: /[+-]?/} //fractional
       // definition: defining word, defined word, etc
-      {regex: /(\:)(\s+)(\S+)(\s+)(\()/, token: ["keyword", null, "def", null, "keyword"], next: "stack"},
+      {regex: /((?:GENERIC)|\:?\:)(\s+)(\S+)(\s+)(\()/, token: ["keyword", null, "def", null, "keyword"], next: "stack"},
+      // method definition: defining word, type, defined word, etc
+      {regex: /(M\:)(\s+)(\S+)(\s+)(\S+)/, token: ["keyword", null, "def", null, "tag"]},
       // vocabulary using --> state
       {regex: /USING\:/, token: "keyword", next: "vocabulary"},
       // vocabulary definition/use
-      {regex: /(USE\:|IN\:)(\s+)(\S+)/, token: ["keyword", null, "variable-2"]},
-      // <constructors>
-      {regex: /<\S+>/, token: "builtin"},
+      {regex: /(USE\:|IN\:)(\s+)(\S+)(?=\s|$)/, token: ["keyword", null, "tag"]},
+      // definition: a defining word, defined word
+      {regex: /(\S+\:)(\s+)(\S+)(?=\s|$)/, token: ["keyword", null, "def"]},
       // "keywords", incl. ; t f . [ ] { } defining words
-      {regex: /;|t|f|if|\.|\[|\]|\{|\}|MAIN:/, token: "keyword"},
+      {regex: /(?:;|\\|t|f|if|loop|while|until|do|PRIVATE>|<PRIVATE|\.|\S*\[|\]|\S*\{|\})(?=\s|$)/, token: "keyword"},
+      // <constructors> and the like
+      {regex: /\S+[\)>\.\*\?]+(?=\s|$)/, token: "builtin"},
+      {regex: /[\)><]+\S+(?=\s|$)/, token: "builtin"},
+      // operators
+      {regex: /(?:[\+\-\=\/\*<>])(?=\s|$)/, token: "keyword"},
       // any id (?)
       {regex: /\S+/, token: "variable"},
 
       {
-        regex: /./,
+        regex: /(?:\s*)|./,
         token: null
       }
     ],
     vocabulary: [
       {regex: /;/, token: "keyword", next: "start"},
-      {regex: /\S+/, token: "variable-2"},
+      {regex: /\S+/, token: "tag"},
       {
-        regex: /./,
+        regex: /(?:\s*)|./,
         token: null
       }
     ],
     string: [
-      {regex: /(?:[^\\]|\\.)*?"/, token: "string", next: "start"},
-      {regex: /.*/, token: "string"}
+        {regex: /(?:[^\\]|\\.)*?"/, token: "string", next: "start"},
+        {regex: /.*/, token: "string"}
+    ],
+    string2: [
+        {regex: /^;/, token: "keyword", next: "start"},
+        {regex: /.*/, token: "string"}
     ],
     string3: [
-      {regex: /(?:[^\\]|\\.)*?"""/, token: "string", next: "start"},
-      {regex: /.*/, token: "string"}
+        {regex: /(?:[^\\]|\\.)*?"""/, token: "string", next: "start"},
+        {regex: /.*/, token: "string"}
     ],
     stack: [
-      {regex: /\)/, token: "meta", next: "start"},
+      {regex: /\)/, token: "keyword", next: "start"},
```

Comment:
This requires a bit of explanation of the language. Factor is a sort of a modern incarnation of Forth. It has no punctuation. In fact, there are no keywords either, but some words fit better as keywords when you want to make more sense of the program. For the most part, those are the defining words - they parse the compiler input. And, Factor allows any symbols in the words except whitespace. And so, many words in standard Factor library look like puntuation.